src/main/java/com/dev/cinema/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package com.dev.cinema.service;
+
+import com.dev.cinema.dao.OrderDao;
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.model.Order;
+import com.dev.cinema.model.Ticket;
+import com.dev.cinema.model.User;
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private OrderDao orderDao;
+    @Inject
+    private ShoppingCartService shoppingCartService;
+
+    @Override
+    public Order completeOrder(List<Ticket> tickets, User user) {
+        Order order = new Order(tickets, user, LocalDateTime.now());
```

Comment:
debug orderDao.add() method and see if it does request to delete tickets from shopping cart too(even though shoppingCartService.clear did it already)

Reply:
fixed